src/main/java/com/fasterxml/jackson/databind/type/ResolvedRecursiveType.java

```diff
@@ -126,16 +126,16 @@ public String toString() {
         return sb.toString();
     }
 
-    @Override
-    public boolean equals(Object o) {
-        if (o == this) return true;
-        if (o == null) return false;
-        if (o.getClass() == getClass()) {
+    //@Override
+    //public boolean equals(Object o) {
+        //if (o == this) return true;
+        //if (o == null) return false;
+        //if (o.getClass() == getClass()) {
             // 16-Jun-2017, tatu: as per [databind#1658], cannot do recursive call since
             //    there is likely to be a cycle...
 
             // but... true or false?
```

Comment:
I left the commented method in place to retain these notes, let me know if you're prefer if they were deleted.

Reply:
Sounds ok to me; just add your own dated note to that effect (left for reference)